src/backend/api/meals.js

```diff
@@ -12,103 +13,157 @@ router.get("/", async(request, response) => {
+                description: request.body.description,
+                location: request.body.location,
+                max_reservations: request.body.max_reservations,
+                when: request.body.when,
+                price: request.body.price,
+            })
+            .where("id", request.params.id)
+            .then(function(result) {
+                response.json({ success: true, message: "ok" }); // respond back to request
+            });
+        //response.json(meals);
     } catch (error) {
-        console.log(error);
-        response.send(error);
+        throw error;
     }
 });
-//Deletes the meal by id
+
 router.delete("/:id", async(request, response) => {
     try {
-        const deleteMealById = await knex("meals")
-            .where("id", parseInt(request.params.id))
-            .del();
-        response.json(`Deleted meal: ${deleteMealById}`);
+        // knex syntax for selecting things. Look up the documentation for knex for further info
+        const meals = await knex("meals")
+            .where("id", request.params.id)
+            .del()
+            .then(function(result) {
+                response.json({ success: true, message: "ok" }); // respond back to request
+            });
+        //response.json(meals);
     } catch (error) {
-        console.log(error);
-        response.send(error);
+        throw error;
     }
 });
 
-//Query. Get meals that has a price smaller than maxPrice
-
-//Query Parameters
-//	Get meals that has a price smaller than maxPrice
 router.get("/", async(request, response) => {
-
-    //     '   if ('
-    //     maxPrice ' in request.query) {
-    //     meals = meals.where('price', '<=', parseInt(request.query.maxPrice))
-    //     response.send()
-    // }
-
-    //Get meals that still has available reservations
-    // if ('availableReservations' in request.query) {
-    //     meals = meals
-    //         .select('meals.id')
-    //         .leftJoin('reservations', { 'meals.id': 'reservations.meal_id' })
-    //         .count({ 'reservations': 'reservations.meal_id' })
-    //         .groupBy('meals.id')
-    //         .having(knex.raw('meals.max_reservations > reservations'));
-    // }
-    //Get meals that partially match a title.
-    // if ('title' in request.query) {
-    //     meals = meals.where("meals.title", "like", `%${request.query.title}%`)
-    // }
-    // //Get meals that has been created after the date	
-    // if ('createdAfter' in request.query) {
-    //     meals = meals.where("meals.createdAfter", ">", request.query.createdAfter);
-    // }
-    // //Only specific number of meals
-    // if ('limit' in request.query) {
-    //     meals = meals.limit(parseInt(request.query.limit));
-    // }
     try {
-        let meals = knex('meals');
 
-        response.json(meals);
+        // knex syntax for selecting things. Look up the documentation for knex for further info
+        const meals = await knex("meals");
+
```

Comment:
try to watch this video and not using if-else too much you can use if multiple times 
https://www.youtube.com/watch?v=EumXak7TyQ0